modules/src/core/ics02_client/client_state.rs

```diff
@@ -257,8 +285,32 @@ impl From<AnyClientState> for Any {
     }
 }
 
-impl ClientState for AnyClientState {
+impl<G> ClientState for AnyClientState<G>
+where
+    G: GlobalDefs + Clone,
+    G::HostFunctions: Sync + Send + Clone + Debug + Eq,
+
+    TendermintConsensusState: TryFrom<ConsensusStateOf<G>, Error = Error>,
+    ConsensusStateOf<G>: From<TendermintConsensusState>,
+
+    BeefyConsensusState: TryFrom<ConsensusStateOf<G>, Error = Error>,
```

Comment:
why do the specific versions of Consensus State need to be added here?

Reply:
Because it's an implementation for `AnyClient`, which depends on every particular client (`beefy`, `near`, etc.)